pemFioi/quickpi/blocklyQuickPi_lib.js

```diff
@@ -7933,8 +7943,43 @@ var getContext = function (display, infos, curLevel) {
             // This needs to be in front of everything
             sensor.focusrect.toFront();
         }
+
+        // save the sensor if we are not running
+        if (!(context.runner && context.runner.isRunning())) {
+            if (_findFirst(sensorDefinitions, function(globalSensor) {
+                return globalSensor.name === sensor.type;
+            }).isSensor) {
+                context.sensorsSaved[sensor.name] = {
+                    state: Array.isArray(sensor.state) ? _copyArray(sensor.state) : sensor.state,
+                    screenDrawing: sensor.screenDrawing,
+                    lastDrawnTime: sensor.lastDrawnTime,
+                    lastDrawnState: sensor.lastDrawnState,
+                    callsInTimeSlot: sensor.callsInTimeSlot,
+                    lastTimeIncrease: sensor.lastTimeIncrease,
+                    removed: sensor.removed,
+                    quickStore: sensor.quickStore
+                };
+            }
+        }
+    }
+
+    function _copyArray(src) {
```

Comment:
You can just use the slice function (`src.slice()`) to do this, no need for a function.